apps/s_server.c

```diff
@@ -2802,8 +2839,20 @@ static int www_body(char *hostname, int s, int stype, unsigned char *context)
 
 #ifdef FIONBIO
     if (s_nbio) {
-        unsigned long sl = 1;
 
+# if defined(OPENSSL_SYS_VMS) && defined(__VMS_VER) && (__VMS_VER >= 70000000)
+        /* For 64-bit --> 32-bit restricted APIs (IOCTL) */
+#  if __INITIAL_POINTER_SIZE == 64
+#   pragma __required_pointer_size __save
+#   pragma __required_pointer_size 32
+#  endif
+        unsigned int sl ;
+#  if __INITIAL_POINTER_SIZE == 64
+#   pragma __required_pointer_size __restore
+#  endif
+# else
```

Comment:
I fail to grasp this. Why the "dance" with pointers? Doesn't BIO_socket_ioctl take care of if? How come long works for all, but not VMS? And why is it uninitialized on VMS?